data/utils/defectsFromBias.py

```diff
@@ -112,13 +46,20 @@ def writeDefectsFile(bboxList, path, detectorSerial, detectorName):
     args = parser.parse_args()
 
     biasMI = afwImage.MaskedImageF(args.bias)
-    defectList = getDefectListFromMask(biasMI, "BAD")
-    bboxList = [defect.getBBox() for defect in defectList]
-    writeDefectsFile(bboxList, DefectsPath, detectorSerial, detectorName)
+    defectList = Defects.fromMask(biasMI, "BAD")
+    valid_start = date_parser.parse('19700101T000000')
+    md = defectList.getMetadata()
+    md['INSTRUME'] = 'test'
+    md['DETECTOR'] = detectorName
+    md['CALIBDATE'] = valid_start.isoformat()
+    md['FILTER'] = None
+    md['CALIB_ID'] = (f'detector={detectorName} calibDate={valid_start.isoformat()} '
+                      'ccd={detectorName} ccdnum={detectorName} filter=None')
+    defectList.writeText(DefectsPath)
     print("wrote defects file %r" % (DefectsPath,))
 
-    test2BBoxList = getBBoxList(DefectsPath, detectorName)
-    assert len(bboxList) == len(test2BBoxList)
-    for boxA, boxB in zip(bboxList, test2BBoxList):
-        assert boxA == boxB
+    test2defectList = Defects.readText(DefectsPath+".ecsv")
```

Comment:
Would it work to include the ".ecsv" suffix on DefectsPath itself -- would the correct file get written? If so, I think it would make the code a bit simpler and easier to understand. Also the description string would give the correct path for the output file (right now it is missing the ".ecsv" suffix).

Reply:
I think you want to get the return value from `writeText` and use that directly in `readText` (writeText forces the file extension based on the type of text file it is writing).  The debug print message should also use the actual name used as returned by writeText.

Reply:
That sounds reasonable, though it means that the full name of the file cannot be included in the help string for the command line parser is produced. I suppose one could always print a message when saving the data.

Reply:
I believe I have addressed this.